README.md

```diff
@@ -2,42 +2,100 @@
 
 ## Applications
 
-Applications are located in bin directory. Before running those, please make
-sure that required Python packages (in python directory) are available:
+Applications are located in bin directory.
+Before running those, please make sure that required Python packages (in python directory) are available:
 
+```bash
 python3.8 -c "from lsst.ts.cRIOpy.GUI import *"
+```
+
+The above command shall pass without error.
+
+### M1M3GUI
+
+GUI/EUI for M1M3 static supports.
+Allows to perform various M1M3 operations, including FA bump testing and manual mirror positioning.
+Requires [ts_m1m3support](https://github.com/lsst-ts/ts_m1m3support) CSC running.
+
+#### Run M1M3GUI with the Simulator
+
+To run the GUI with the CSC in simulation mode under the docker container, you can build the docker image of CSC first by:
+
+```bash
+git clone https://github.com/lsst-ts/ts_m1m3support.git
+cd ts_m1m3support
+docker build -t m1m3sim .
+```
+
+Enter the container with the display setting (use Mac OS as an example):
+
+```bash
+xhost +
+IP=$(ifconfig en0 | grep inet | awk '$1=="inet" {print $2}')
+docker run -it --rm -e DISPLAY=${IP}:0 -v ${path_to_local_ts_cRIOpy}:/home/saluser/ts_cRIOpy -v /tmp/.X11-unix:/tmp/.X11-unix m1m3sim:latest
+```
 
-shall pass without error.
+Please note XQuartz settings should allow any client: [Running GUI’s with Docker on Mac OS X](https://cntnr.io/running-guis-with-docker-on-mac-os-x-a14df6a76efc).
 
-### M1M3GUI 
+Inside the container, do:
+
+```
+conda install -y pyside2 asyncqt h5py
+cd ts_cRIOpy/
+setup -kr .
+export LSST_DDS_PARTITION_PREFIX=test
+cd bin/
+M1M3GUI
+```
+
+You have to set **LSST_DDS_PARTITION_PREFIX** variable to match your environment. 
+
+After the EUI is running, you can use another terminal session to enter the same container by:
+
+```bash
+docker exec -it ${container_id} bash
+```
+
+
+Built the M1M3 simulator by following: [ts_m1m3support](https://github.com/lsst-ts/ts_m1m3support/blob/main/README.md).
+Inside the container, do:
+
+```bash
+source ~/.setup_dev.sh
+cd repos/ts_m1m3support
+export LSST_DDS_PARTITION_PREFIX=test
+./ts-M1M3supportd -c SettingFiles/ -f
+```
 
-GUI/EUI for M1M3 static supports. Allows to perform various M1M3 operations,
-including FA bump testing and manual mirror positioning. Requires
-[ts_m1m3support](https://github.com/lsst-ts/ts_m1m3support) CSC running.
+**LSST_DDS_PARTITION_PREFIX** variable in simulator must match EUI settings.
+Please run the EUI first.
+Only after EUI is running, run the simulator.
+You can then restart the simulator as needed.
 
 ### M1M3TSGUI
 
-GUI/EUI for M1M3 thermal system. Allows TS monitoring and performing various
```

Comment:
Can you please keep the original, 70 char wide wrapping? There ins't need to break by lines, I know how to reformat text in Vim to fit block with, and it makes editing much easier than those long lines showing later,